docs/src/main/asciidoc/spring-cloud-gateway.adoc

```diff
@@ -2534,6 +2536,16 @@ public class PostGatewayFilterFactory extends AbstractGatewayFilterFactory<PostG
 ----
 ====
 
+==== Naming Custom Filter And Referencing Them In Configuration
+
+A filter named `Something` should be implemented by a
+`SomethingGatewayFilterFactory` class and referenced as `Something` in
+configuration files.
+
+Alternatively, a gateway filter factory named `AnotherThing` (without the)
```

Comment:
This sentence seems to just repeat the first and seems redundant.

Reply:
I'm afraid not:
The first sentence documents the case where the suffix GatewayFilterFactory is added automatically to the class name from the filter name.
The second one document the case where the filter and its class have exactly the same name.

Thus, I tried to make this sentence more explicit.

Reply:
That is an undocumented thing that happens to work. I don't want to document it.

Reply:
I understand your point of view, but as it works and as the proper way of declaring filters is currently undocumented, this exists in code in production now. To be precise, this is exactly the point that made me issue #1589 at the fist place.
Maybe I can add a warning this must not be done and must be fixed if already exists, as this is not a feature and is not guarantee to work in future releases?

Reply:
Here I improved this paragraph. In my opinion, there should be a mention of this syntax, at least to warn users to remove it, as it will be a de facto breaking change if it is used when this syntax will be removed from code.